versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/objtypes/Hashes.java

```diff
@@ -122,4 +123,10 @@ static ObjId stringDataHash(
     hasher.putBytes(text.asReadOnlyByteBuffer());
     return hashAsObjId(hasher);
   }
+
+  static ObjId uniqueIdHash(String space, String value) {
```

Comment:
One general remark: wouldn't it be simpler to model the value as an opaque byte array? It seems we could save some conversions to and from string in the common case where the value is an UUID.